### PR TITLE
Fixes #2478 and implements system test for FSecBugs #2479

### LINKED ISSUE #2478: PDS-FindSecurityBugs introduce JVM memory setting
Introduce the ability to set the memory of the JVM for FindSecurityBugs.

### PR DESCRIPTION
**Pull request by Jeeppler**
- fixes #2478 by enforcing PDS_MAX_FILE_UPLOAD_BYTES, regardless whether the file comes from a file share or is getting uploaded
- implements system test #2479
- refactoring and adjusting code to improve the code quality

closes: #2478
closes: #2479

### CODE REVIEW

File: sechub-pds-solutions/findsecuritybugs/docker/pds-config.json [hunk 1]
```diff
@@ -1,58 +1,51 @@
...
-            "key": "findsecuritybugs.include.bugpatterns",
-            "description": "Include only the given bug patterns. A list of bug patter names can be found at: https://find-sec-bugs.github.io/bugs.htm. For example: WEAK_MESSAGE_DIGEST_MD5, SQL_INJECTION."
-          },
-          {
-            "key": "findsecuritybugs.exclude.bugpatterns",
-            "description": "Exclude the given bug patterns. A list of bug patter names can be found at: https://find-sec-bugs.github.io/bugs.htm. For example: WEAK_MESSAGE_DIGEST_MD5, SQL_INJECTION."
-          }
-        ]
-      }
+  "apiVersion" : "1.0",
```
Comment (sven-dmlr) [inline sechub-pds-solutions/findsecuritybugs/docker/pds-config.json:2]: Was there changed anything int his file beside the formatting?
(I find the old formatting easier to read)

-> Should we add a section for JSON to the coding conventions? https://mercedes-benz.github.io/sechub/latest/sechub-techdoc.html#section-coding-conventions

File: sechub-pds-solutions/findsecuritybugs/env [hunk 2]
```diff
@@ -1,7 +1,2 @@
-# The base image to use
-# uncomment to use local image
-# BASE_IMAGE="pds-base_pds"
-BASE_IMAGE="ghcr.io/mercedes-benz/sechub/pds-base"
```
Comment (sven-dmlr) [inline sechub-pds-solutions/findsecuritybugs/env:4]: Please leave this in! It is needed for the release builds.

File: sechub-pds-solutions/findsecuritybugs/tests/download_test_files.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: MIT
+
+current_test_folder="$1"
+test_file="$2"
+test_folder_name="$3"
+	
+if [[ ! -d "$current_test_folder" ]]
+then
+	echo "Target folder is empty"
```
Comment (sven-dmlr) [inline sechub-pds-solutions/findsecuritybugs/tests/download_test_files.sh:10]:
```suggestion
	echo "Target folder \"$current_test_folder\" does not exist"
```

File: sechub-pds-solutions/findsecuritybugs/tests/download_test_files.sh [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+if [[ -z "$test_file" ]]
+then
+	echo "No file to test provided"
+	exit 1
+fi
+
+if [[ -z "$test_folder_name" ]]
+then
+	echo "No file to test provided"
```
Comment (sven-dmlr) [inline sechub-pds-solutions/findsecuritybugs/tests/download_test_files.sh:22]: Same text as in line 16. Is this correct?
The variable name sounds like a folder and not like a file.

**Review by sven-dmlr — CHANGES_REQUESTED** [review]
Looks good overall.
Please check my comments.

File: sechub-pds-solutions/findsecuritybugs/docker/pds-config.json (hunk 1 above)
Comment (Jeeppler) [reply sechub-pds-solutions/findsecuritybugs/docker/pds-config.json:2]: Only formatting changes.

I used the formatter of the HiJSON Eclipse Plugin developed by @de-jcup.

We probably should add a section for a JSON coding convention.

**Review by sven-dmlr — APPROVED** [review]
:+1:

**Pull request merged**